### ISSUE TITLE
prevent structures from generating on world import

### ISSUE THREAD

**Issue by AvatarAcid** [Multiverse/Multiverse-Core#1610]
Hi MultiVerse,

I found this link:
https://github.com/Multiverse/Multiverse-Core/wiki/Command-Reference#wiki-mapfeatures

I ran the following command for import:
/mv import world_name normal -a false

The world imported with no issues, however structures are destroying pre-built things in my world. There is a picture here:
https://www.dropbox.com/s/o9ssuvxx913oxlp/mv-structures.png?dl=0

I am running Craftbukkit/Spigot v1.8.8 and Multiverse-Core Version: 2.5-b699

Can you please provide the proper method of importing a pre-existing world and not allowing structures to generate within?

Thanks!

**Comment by AvatarAcid** [comment]
Just in case; the structures are not in the world I'm importing already :)

**Comment by AvatarAcid** [comment]
>mv import world_pvp normal -a FALSE
[18:19:09 INFO]: CONSOLE: Starting import of world 'world_pvp'...
[18:19:09 INFO]: [Multiverse-Core] Loading World & Settings - 'world_pvp' - Env:
 NORMAL - Type: null
[18:19:09 INFO]: -------- World Settings For [world_pvp] --------
[18:19:09 INFO]: Arrow Despawn Rate: 1200
[18:19:09 INFO]: Allow Zombie Pigmen to spawn from portal blocks: true
[18:19:09 INFO]: Nerfing mobs spawned from spawners: false
[18:19:09 INFO]: Cactus Growth Modifier: 100%
[18:19:09 INFO]: Cane Growth Modifier: 100%
[18:19:09 INFO]: Melon Growth Modifier: 100%
[18:19:09 INFO]: Mushroom Growth Modifier: 100%
[18:19:09 INFO]: Pumpkin Growth Modifier: 100%
[18:19:09 INFO]: Sapling Growth Modifier: 100%
[18:19:09 INFO]: Wheat Growth Modifier: 100%
[18:19:09 INFO]: NetherWart Growth Modifier: 100%
[18:19:09 INFO]: Entity Activation Range: An 32 / Mo 32 / Mi 16
[18:19:09 INFO]: Sending up to 10 chunks per packet
[18:19:09 INFO]: Max Entity Collisions: 8
[18:19:09 INFO]: Custom Map Seeds:  Village: 10387312 Feature: 14357617
[18:19:09 INFO]: Item Merge Radius: 2.5
[18:19:09 INFO]: Item Despawn Rate: 6000
[18:19:09 INFO]: Chunks to Grow per Tick: 650
[18:19:09 INFO]: Clear tick list: false
[18:19:09 INFO]: Experience Merge Radius: 3.0
[18:19:09 INFO]: Anti X-Ray: true
[18:19:09 INFO]:        Engine Mode: 1
[18:19:09 INFO]:        Hidden Blocks: [14, 15, 16, 21, 48, 49, 54, 56, 73, 74,
82, 129, 130]
[18:19:09 INFO]:        Replace Blocks: [1, 5]
[18:19:09 INFO]: Mob Spawn Range: 4
[18:19:09 INFO]: Entity Tracking Range: Pl 48 / An 48 / Mo 48 / Mi 32 / Other 64

[18:19:09 INFO]: Hopper Transfer: 8 Hopper Check: 8 Hopper Amount: 1
[18:19:09 INFO]: Random Lighting Updates: false
[18:19:09 INFO]: Structure Info Saving: true
[18:19:09 INFO]: Max TNT Explosions: 100
[18:19:09 INFO]: Tile Max Tick Time: 50ms Entity max Tick Time: 50ms
[18:19:09 INFO]: Zombie Aggressive Towards Villager: true
[18:19:09 INFO]: View Distance: 15
[18:19:09 INFO]: Preparing start region for level 7 (Seed: 27594263)
[18:19:10 INFO]: [WorldGuard] Default configuration file written: C:\_bukkits\burrows_spigot_noadmin\MineCraft\plugins\WorldGuard\worlds\world_pvp\config.yml
[18:19:10 INFO]: [WorldGuard] Default configuration file written: C:\_bukkits\burrows_spigot_noadmin\MineCraft\plugins\WorldGuard\worlds\world_pvp\blacklist.txt
[18:19:10 INFO]: [WorldGuard] (world_pvp) TNT ignition is PERMITTED.
[18:19:10 INFO]: [WorldGuard] (world_pvp) Lighters are PERMITTED.
[18:19:10 INFO]: [WorldGuard] (world_pvp) Lava fire is blocked.
[18:19:10 INFO]: [WorldGuard] (world_pvp) All fire spread is disabled.
[18:19:10 INFO]: [WorldGuard] Loaded configuration for world 'world_pvp'
[18:19:10 INFO]: [dynmap] Loaded 3 maps of world 'world_pvp'.
[18:19:10 INFO]: CONSOLE: Complete!
>

**Comment by TWSSYesterday** [comment]
Import the world if it already has structures denied, if you change it when it was made with structures, it can cause issues.

**Comment by AvatarAcid** [comment]
Thanks for much for your reply :) 

The world I'm attempting to import was generated by WorldPainter which I think still lacks the option not to generate structures.

So the work around for this if anyone else would like to know:
Google and download NBTExplorer. Open your level.dat and change the option "MapFeatures" from a 1 to a 0 and then run a normal MV IMPORT:
mv import worldname normal

After that process, I get the following back from mv info:
[12:27:38 INFO]: Structures: false

That is great, I tested and the world is now imported via multiverse and is not generating structures all over my beautiful world :P 

It might be good to update the mv import doco with the fact that importing and switching that setting in the world may not be supported. Especially since it wasn't changing that MapFeautres flag at all when importing a world.

I did go back and test the -a FALSE option on creating a new world and you are right, multiverse handled that brilliantly :)
